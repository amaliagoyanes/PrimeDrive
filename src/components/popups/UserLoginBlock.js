import React, {Component} from 'react';
import {
    Text, View
} from 'react-native';
import {Row, Col} from 'react-native-easy-grid';
import * as Progress from 'react-native-progress';
import {ButtonItem, NumericKeyboard} from '../index';
import styles from '../../style/styles';
import {scale, scaleText} from '../../scaling';


export default class UserLoginBlock extends Component {
    constructor(props) {
        super(props);
        this.state = {
            input: '', error: false
        };

        this.numOnPress = this.numOnPress.bind(this);
    }

    numOnPress(text) {
        this.setState({input: this.state.input + text, error: false})
    }

    render() {
        let input = this.state.input;
        if (this.props.processing) {
            return (
                <Col size={4} style={styles.userLoginRight}>
                    <View style={{flexDirection: 'row', justifyContent: 'center', flex: 1, alignItems: 'center'}}>
                        <Progress.Circle borderWidth={5} borderColor={"#2dab61"} size={scale(60)} indeterminate={true} />
                    </View>
                </Col>
            )
        }
        return (
            <Col size={4} style={styles.userLoginRight}>
                <Row size={1} style={{justifyContent: 'center', alignItems: 'center'}}>
                    {this.state.error ?
                        <Text style={[styles.text, {color: '#e84b3a'}]}>
                            {this.props.intl !== undefined &&
                            this.props.intl.messages["app.noUser"]
                            }</Text> :
                        <Col/>
                    }
                </Row>
                <NumericKeyboard size={6.5} numOnPress={this.numOnPress} input={input} intl={this.props.intl}
                                 cleanInput={() => this.setState({input: input.slice(0, input.length - 1)})}/>
                <Row size={1.25}>
                    <ButtonItem text="app.login" onPress={() => this.props.login(this.state.input)} intl={this.props.intl}
                                style={[styles.modalButton, {backgroundColor: '#3cb671', marginRight: 0}]}
                                textStyle={styles.modalButtonText}/>
                </Row>
            </Col>
        )
    }
}
