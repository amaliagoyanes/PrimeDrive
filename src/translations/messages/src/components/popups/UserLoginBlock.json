[
  {
    "id": "app.noUser",
    "defaultMessage": "Such user doesn't exist"
  }
]