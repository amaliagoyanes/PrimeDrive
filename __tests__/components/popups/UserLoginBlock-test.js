'use strict';

import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import UserLoginBlock from '../../../src/components/popups/UserLoginBlock';


it('renders correctly', () => {
    const tree = shallow(
        <UserLoginBlock />);
    expect(tree).toMatchSnapshot();
    expect(tree.find('NumericKeyboard').length).toEqual(1);
    expect(tree.find('ButtonItem').length).toEqual(2);
});

