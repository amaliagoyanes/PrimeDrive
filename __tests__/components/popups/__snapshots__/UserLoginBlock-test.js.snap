// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ColumnNB
  size={4}
  style={
    Object {
      "marginBottom": 20,
      "paddingLeft": 30,
      "paddingRight": 25,
    }
  }
>
  <RowNB
    size={1}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <ColumnNB />
  </RowNB>
  <NumericKeyboard
    cleanInput={[Function]}
    input=""
    numOnPress={[Function]}
    size={6}
  />
  <RowNB
    size={1}
  >
    <ButtonItem
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginLeft": 0,
            "marginRight": 40,
          },
          Object {
            "backgroundColor": "#3cb671",
            "marginRight": 0,
          },
        ]
      }
      text="Login"
      textStyle={
        Object {
          "color": "#fefefe",
          "fontSize": 16,
        }
      }
    />
  </RowNB>
  <RowNB
    size={1}
  >
    <ButtonItem
      onPress={[Function]}
      style={
        Array [
          Object {
            "marginLeft": 0,
            "marginRight": 40,
          },
          Object {
            "backgroundColor": "#bec3c7",
            "marginRight": 0,
          },
        ]
      }
      text="Switch Off"
      textStyle={
        Object {
          "color": "#fefefe",
          "fontSize": 16,
        }
      }
    />
  </RowNB>
</ColumnNB>
`;
